cpp/src/arrow/adapters/orc/util.h

```diff
@@ -35,8 +35,9 @@ Result<std::shared_ptr<DataType>> GetArrowType(const liborc::Type* type);
 
 Result<ORC_UNIQUE_PTR<liborc::Type>> GetOrcType(const Schema& schema);
 
-Status AppendBatch(const liborc::Type* type, liborc::ColumnVectorBatch* batch,
-                   int64_t offset, int64_t length, arrow::ArrayBuilder* builder);
+ARROW_EXPORT Status AppendBatch(const liborc::Type* type,
```

Comment:
I have to mark them as `ARROW_EXPORT`. Otherwise the windows build will complain unresolved references for the new tests.